registry/remote/repository.go

```diff
@@ -99,6 +94,11 @@ type Repository struct {
 	// list, and referrers list.
 	// If less than or equal to zero, a default (currently 4MiB) is used.
 	MaxMetadataBytes int64
+
+	// referrersUnsupported indicates if Referrers API is unsupported for the
+	// repository.
+	//  0: false (supported), 1: true (unsupported).
+	referrersUnsupported int32
```

Comment:
`oras-go` should provide the ability to pre-set whether the remote server supports referrers API without access it.

Basically, it can have 3 states: unknown (detection required), known that remote server has support, and known that remote server has no support on referrers API. Let's discuss this part in detail.

Reply:
Updated.